pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -3185,3 +3209,48 @@ func Test_storeTaskSpec(t *testing.T) {
 		t.Fatalf(diff.PrintWantGot(d))
 	}
 }
+
+func TestGetResultsShouldFilterInternalTektonResultType(t *testing.T) {
+
+	input := []v1beta1.PipelineResourceResult{
+		v1beta1.PipelineResourceResult{
+			Key:        "",
+			Value:      "",
+			ResultType: v1beta1.InternalTektonResultType,
+		},
+		v1beta1.PipelineResourceResult{
+			Key:        "",
+			Value:      "",
+			ResultType: v1beta1.TaskRunResultType,
+		},
+		v1beta1.PipelineResourceResult{
+			Key:        "",
+			Value:      "",
+			ResultType: v1beta1.PipelineResourceResultType,
+		},
+	}
+
+	expectedTaskRunResults := []v1beta1.TaskRunResult{
+		v1beta1.TaskRunResult{
+			Name:  "",
+			Value: "",
+		},
+	}
+	expectedPipelineResourceResults := []v1beta1.PipelineResourceResult{
+		input[2],
+	}
+
+	gotTaskRunResults, gotPipelineResourceResults := getResults(input)
```

Comment:
i'm also wondering about testing this private method - often testing private methods is a sign that the method under test could be factored differently, e.g. be an exported part of a different package, or that the testing belongs at a different level (e.g. the thing calling getResults)

im usually more inclined to the first option :D

Reply:
Please see #3138 for a cleanup.